tests/plugins/test_antpath.py

```diff
@@ -0,0 +1,59 @@
+# -*- coding: utf-8 -*-
+
+"""
+Test AntPath
+-------------
+"""
+
+from __future__ import (absolute_import, division, print_function)
+
+import folium
+from folium import plugins
+
+from jinja2 import Template
+
+
+def test_antpath():
+    m = folium.Map([20., 0.], zoom_start=3)
+
+    locations = [
+        [59.355600, -31.99219],
+        [55.178870, -42.89062],
+        [47.754100, -43.94531],
+        [38.272690, -37.96875],
+        [27.059130, -41.13281],
+        [16.299050, -36.56250],
+        [8.4071700, -30.23437],
+        [1.0546300, -22.50000],
+        [-8.754790, -18.28125],
+        [-21.61658, -20.03906],
+        [-31.35364, -24.25781],
+        [-39.90974, -30.93750],
+        [-43.83453, -41.13281],
+        [-47.75410, -49.92187],
+        [-50.95843, -54.14062],
+        [-55.97380, -56.60156]
+    ]
+
+    antpath = plugins.AntPath(locations=locations)
+    antpath.add_to(m)
+
+    m._repr_html_()
+    out = m._parent.render()
+
+    # We verify that the script import is present.
+    script = '<script src="https://cdn.jsdelivr.net/npm/leaflet-ant-path@1.1.2/dist/leaflet-ant-path.min.js"></script>'  # noqa
+    assert script in out
+
+    # We verify that the script part is correct.
+    tmpl = Template("""
+          {{this.get_name()}} = L.polyline.antPath(
+                  {{this.location}},
+                  {{ this.options }}
+                )
+                .addTo({{this._parent.get_name()}});
+        """)  # noqa
+
+    expected_rendered = tmpl.render(this=antpath)
+    rendered = antpath._template.module.script(antpath)
+    assert folium.utilities.compare_rendered(expected_rendered, rendered)
```

Comment:
I'm planning on a major test re-factor soon.
Probably using this approach if you all agree on it.

Reply:
I do like standardizing the comparison function. But I'm not sure this way of testing is that useful. We only test if we correctly copied the template definition here correctly. Suppose the rendered javascript is incorrect, it will be incorrect for both `rendered` and `expected_rendered`. If we really want to do a functional test, we should store the rendered html + js that we know is correct and compare the output of `rendered` with that.

I would suggest discussing a test refactor separately so we can get this PR merged.

Reply:
Indeed. We need that kind of testing since the beginning of `folium` :grimacing: 

> I would suggest discussing a test refactor separately so we can get this PR merged.

:+1: